cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -7,50 +14,93 @@ module Cardano.CLI.Shelley.Run.Query
 
 import           Cardano.Crypto.Hash.Class (getHashBytesAsHex)
+import           Network.Mux (MuxMode(..), MuxTrace, WithMuxBearer)
+import           Ouroboros.Consensus.Block (CodecConfig)
 
 import           Ouroboros.Consensus.Cardano (protocolClientInfo)
+import           Ouroboros.Consensus.Ledger.Abstract (Query)
 import           Ouroboros.Consensus.Node.ProtocolInfo (pClientInfoCodecConfig)
+import           Ouroboros.Consensus.Node.Run (RunNode)
+import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock)
 import           Ouroboros.Consensus.Shelley.Protocol.Crypto (TPraosStandardCrypto)
-import           Ouroboros.Network.NodeToClient (withIOManager)
+import           Ouroboros.Network.Block (Point, getTipPoint)
+import           Ouroboros.Network.NodeToClient (ConnectionId, DictVersion, Handshake,
+                   LocalAddress, NetworkConnectTracers (..), NodeToClientProtocols (..),
+                   NodeToClientVersion, NodeToClientVersionData (..), TraceSendRecv,
+                   Versions, withIOManager)
+import qualified Ouroboros.Network.NodeToClient as NodeToClient
 
-import           Ouroboros.Network.Block (getTipPoint)
 
+import qualified Shelley.Spec.Ledger.Address as Ledger (RewardAcnt (..))
 import           Shelley.Spec.Ledger.Coin (Coin (..))
 import           Shelley.Spec.Ledger.Delegation.Certificates (PoolDistr(..))
+import qualified Shelley.Spec.Ledger.Delegation.Certificates as Ledger (PoolDistr(..))
 import           Shelley.Spec.Ledger.Keys (Hash, KeyHash(..), KeyRole (..), VerKeyVRF)
 import           Shelley.Spec.Ledger.LedgerState (EpochState)
+import qualified Shelley.Spec.Ledger.LedgerState as Ledger
 import           Shelley.Spec.Ledger.PParams (PParams)
-import           Shelley.Spec.Ledger.TxData (TxId (..), TxIn (..), TxOut (..))
-import           Shelley.Spec.Ledger.UTxO (UTxO (..))
+import qualified Shelley.Spec.Ledger.TxData as Shelley (TxId (..), TxIn (..), TxOut (..))
+import qualified Shelley.Spec.Ledger.UTxO as Ledger (UTxO(..))
+
+import           Ouroboros.Consensus.Network.NodeToClient
+import           Ouroboros.Consensus.Node.NetworkProtocolVersion
+                  (nodeToClientProtocolVersion, supportedNodeToClientVersions)
+import           Ouroboros.Consensus.Shelley.Ledger
+
+import           Ouroboros.Network.Block (Serialised (..))
+import           Ouroboros.Network.Mux (OuroborosApplication(..),
+                   MuxPeer(..), RunMiniProtocol(..))
+import           Ouroboros.Network.Protocol.LocalStateQuery.Client
+                   (ClientStAcquired (..), ClientStAcquiring (..), ClientStIdle (..),
+                    ClientStQuerying (..), LocalStateQueryClient(..), localStateQueryClientPeer)
+import           Ouroboros.Network.Protocol.LocalStateQuery.Type (AcquireFailure (..))
+
+
+
+import           Cardano.Binary (decodeFull)
+import           Cardano.BM.Data.Tracer (ToLogObject (..), nullTracer)
+import           Cardano.BM.Trace (Trace, appendName, logInfo)
```

Comment:
Do we really need these? This is what pulls in the full monitoring framework. Can we stick to just `Control.Tracer` which needs only `contra-tracer`?